pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -1719,6 +1728,42 @@ public RealtimeSegmentDataManager(SegmentZKMetadata segmentZKMetadata, TableConf
     }
   }
 
+  private boolean allowConsumptionDuringCommit() {
+    if (_realtimeTableDataManager.isDedupEnabled()) {
```

Comment:
@klsince does simply adding this solve the inconsistency issue with dedup during segment download or not?

Reply:
I believe so. cc @deepthi912 who is fixing this in a separate short PR and aim to land asap. thanks for catching this!